widget/progressbar.go

```diff
@@ -74,10 +79,13 @@ func (p *progressRenderer) Objects() []fyne.CanvasObject {
 }
 
 // ProgressBar widget creates a horizontal panel that indicates progress
+// An infinite ProgressBar loops 0% -> 100% until StopInfiniteProgress() is called
 type ProgressBar struct {
 	baseWidget
 
-	Min, Max, Value float64
+	Min, Max, Value      float64
+	Infinite             bool
+	stopInfiniteLoopChan chan bool
```

Comment:
very slightly more efficient - can use the (ugly) empty struct channel as a signalling mechanism, where you dont care what the actual value is.

In this case, the consumer of the channel doesnt care if the value is true of false, so its a good candidate for this hack.

```
stopInfiniteLoopChan chan struct{}

then on 128 below

p.stopInfiniteLoopChan <- struct{}{}
```

Reply:
I ended up restructuring a little, but cool to know about the empty signal :)